test/integration/user_muting_test.rb

```diff
@@ -0,0 +1,31 @@
+require "test_helper"
+
+class UserMutingTest < ActionDispatch::IntegrationTest
```

Comment:
This should perhaps be a system test rather than an integration test, where we can use more natural things like "click_button" rather than `assert_select` followed by directly posting to controllers.